common/src/main/java/com/microsoft/identity/common/internal/platform/DeviceKeyManager.java

```
@@ -0,0 +1,191 @@
+                    TAG,
+                    "Error while getting creation date for alias " + mKeyAlias,
+                    e
+            );
+            throw new ClientException(ClientException.KEYSTORE_NOT_INITIALIZED, e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public boolean clear() {
+        boolean deleted = false;
+
+        try {
+            mKeyStore.deleteEntry(mKeyAlias);
+            deleted = true;
+        } catch (final KeyStoreException e) {
+            Logger.error(
+                    TAG,
+                    "Error while clearing KeyStore",
+                    e
+            );
+        }
+
+        return deleted;
+    }
+
+
+    /**
+     * Retrieve the entry stored in this particular alias.
+     * @return the Entry in question, or null if the entry does not exist.
+     * @throws UnrecoverableEntryException
+     * @throws NoSuchAlgorithmException
+     * @throws KeyStoreException
+     */
+    @Override
+    public K getEntry() throws UnrecoverableEntryException, NoSuchAlgorithmException, KeyStoreException {
+        return (K) mKeyStore.getEntry(mKeyAlias, null);
+    }
+
+    @Override
+    public void importKey(byte[] jwk, String algorithm) throws ClientException {
+        final SecretKeySpec key = new SecretKeySpec(jwk, algorithm);
+        final String errCode;
+        final Exception exception;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            try {
+                mKeyStore.setEntry(mKeyAlias, new KeyStore.SecretKeyEntry(key),
+                        new KeyProtection.Builder(KeyProperties.PURPOSE_DECRYPT | KeyProperties.PURPOSE_ENCRYPT |
+                                KeyProperties.PURPOSE_SIGN).build());
+                return;
+            } catch (KeyStoreException e) {
+              errCode = ClientException.KEYSTORE_NOT_INITIALIZED;
+              exception = e;
+            }
+            throw new ClientException(errCode, exception.getMessage());
+        }
```

This needs thought.  Should probably throw unsupported for now.